alfresco-transformer-base/src/test/java/org/alfresco/transformer/AbstractTransformerControllerTest.java

```diff
@@ -162,17 +162,26 @@ void generateTargetFileFromResourceFile(String actualTargetExtension, File testF
 
     protected File getTestFile(String testFilename, boolean required) throws IOException
     {
+        if (System.getProperty("os.name").substring(0, 3).equals("Win"))
+        {
             File testFileUrl = null;
             try {
                 testFileUrl = new File(new File(getClass().getProtectionDomain().getCodeSource().getLocation().toURI()) + "\\" + testFilename);
```

Comment:
In windows implementation the original error handling is not executed. When a file is not found a different error is being thrown for windows and is causing test failures.

Reply:
Found the test failures. It wasn't actually due to the error handling or the line above, it was due to the logic the files are generated through FileChannel which expects null if the file doesn't exist for files which isRequired is set to false, which in the original code is handled by `return testFileUrl == null ? null : new File(testFileUrl.getFile());` 
I also added the exception from the original code for completeness (Although I think that's already handled)